switch_model/generators/core/dispatch.py

```diff
@@ -495,83 +496,86 @@ def c(func):
-        graph_curtailment_per_tech(tools)
-        graph_total_dispatch(tools)
-        graph_hourly_dispatch(tools)
-        graph_hourly_curtailment(tools)
-
-def compare(tools):
-    if not tools.skip_long:
-        if tools.num_scenarios > 1:
-            compare_hourly_dispatch(tools)
-            compare_hourly_curtailment(tools)
-
+@graph(
+    "dispatch",
+    title="Average daily dispatch"
+)
 def graph_hourly_dispatch(tools):
     """
     Generates a matrix of hourly dispatch plots for each time region
     """
     # Read dispatch.csv
     df = tools.get_dataframe('dispatch.csv')
     # Convert to GW
-    df["DispatchGen_GW"] = df["DispatchGen_MW"] / 1000
+    df["DispatchGen_MW"] /= 1000
     # Plot Dispatch
-    tools.graph_time_matrix(df, "DispatchGen_GW",
-                      out="dispatch",
-                      title="Average daily dispatch",
-                      ylabel="Average daily dispatch (GW)")
-
+    tools.graph_time_matrix(
+        df,
+        value_column="DispatchGen_MW",
+        ylabel="Average daily dispatch (GW)",
+    )
 
+@graph(
+    "curtailment",
+    title="Average daily curtailment"
+)
 def graph_hourly_curtailment(tools):
     # Read dispatch.csv
     df = tools.get_dataframe('dispatch.csv')
     # Keep only renewable
     df = df[df["is_renewable"]]
-    df["Curtailment_GW"] = df["Curtailment_MW"] / 1000
+    df["Curtailment_MW"] /= 1e3 # Convert to GW
     # Plot curtailment
     tools.graph_time_matrix(
         df,
-        "Curtailment_GW",
-        out="curtailment",
-        title="Average daily curtailment",
+        value_column="Curtailment_MW",
         ylabel="Average daily curtailment (GW)"
     )
 
-def compare_hourly_dispatch(tools):
+
+@graph(
+    "dispatch_per_scenario",
+    title="Average daily dispatch",
+    requires_multi_scenario=True
+)
+def graph_hourly_dispatch(tools):
     """
     Generates a matrix of hourly dispatch plots for each time region
     """
     # Read dispatch.csv
-    df = tools.get_dataframe('dispatch.csv', all_scenarios=True)
+    df = tools.get_dataframe('dispatch.csv')
     # Convert to GW
-    df["DispatchGen_GW"] = df["DispatchGen_MW"] / 1000
+    df["DispatchGen_MW"] /= 1e3
     # Plot Dispatch
     tools.graph_scenario_matrix(
         df,
-        "DispatchGen_GW",
-        out="dispatch_per_scenario",
-        title="Average daily dispatch",
+        value_column="DispatchGen_MW",
         ylabel="Average daily dispatch (GW)"
     )
 
 
-def compare_hourly_curtailment(tools):
+@graph(
+    "curtailment_compare_scenarios",
+    title="Average daily curtailment by scenario",
+    requires_multi_scenario=True
+)
+def graph_hourly_curtailment(tools):
     # Read dispatch.csv
-    df = tools.get_dataframe('dispatch.csv', all_scenarios=True)
+    df = tools.get_dataframe('dispatch.csv')
     # Keep only renewable
     df = df[df["is_renewable"]]
-    df["Curtailment_GW"] = df["Curtailment_MW"] / 1000
+    df["Curtailment_MW"] /= 1e3
```

Comment:
If you this, wouldn't it be "Curtailment_GW" ?

Reply:
I didn't want to redefine another column and use more memory.

Reply:
That makes sense. Just to make sure it gets rename after?